EKF/control.cpp

```diff
@@ -801,36 +801,29 @@ void Ekf::controlHeightSensorTimeouts()
 			const bool gps_hgt_accurate = (gps_init.vacc < _params.req_vacc);
 
 			const baroSample &baro_init = _baro_buffer.get_newest();
-			bool baro_hgt_available = ((_time_last_imu - baro_init.time_us) < 2 * BARO_MAX_INTERVAL);
+			const bool baro_data_available = ((_time_last_imu - baro_init.time_us) < 2 * BARO_MAX_INTERVAL);
 
 			// check for inertial sensing errors in the last 10 seconds
 			const bool prev_bad_vert_accel = (_time_last_imu - _time_bad_vert_accel < BADACC_PROBATION);
 
 			// reset to GPS if adequate GPS data is available and the timeout cannot be blamed on IMU data
-			bool reset_to_gps = !_gps_hgt_intermittent && gps_hgt_accurate && !prev_bad_vert_accel;
-
-			// reset to GPS if GPS data is available and there is no Baro data
-			reset_to_gps = reset_to_gps || (!_gps_hgt_intermittent && !baro_hgt_available);
-
-			// reset to Baro if we are not doing a GPS reset and baro data is available
-			bool reset_to_baro = !reset_to_gps && baro_hgt_available;
+			const bool reset_to_gps = !_gps_hgt_intermittent &&
+					    ( (gps_hgt_accurate && !prev_bad_vert_accel) || !baro_data_available );
 
 			if (reset_to_gps) {
 				// set height sensor health
 				_baro_hgt_faulty = true;
 
-				// reset the height mode
 				setControlGPSHeight();
 
 				// request a reset
 				reset_height = true;
 				ECL_WARN_TIMESTAMPED("baro hgt timeout - reset to GPS");
 
-			} else if (reset_to_baro) {
+			} else if (baro_data_available) {
```

Comment:
you can also remove all the 4
```c++
} else {
   // we have nothing we can reset to
   // deny a reset
   reset_height = false;
}
```
as the reset_height variable is already initialized to `false`

Reply:
Nice idea